spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourceSpecTest.java

```diff
@@ -901,6 +903,71 @@ public void shouldMigrateAllConsumersOfAnonymousOwnerToTargetOwner() {
             .doesNotContainAnyElementsOf(oldScaCerts);
     }
 
+    @Test
+    public void shouldRevokeEntitlementsAndRemoveActivationKeyPoolsWithChangeToSCA()
+        throws InterruptedException {
+        ApiClient admin = ApiClients.admin();
+        OwnerDTO owner = admin.owners().createOwner(Owners.random());
+
+        ProductDTO prod = admin.ownerProducts().createProduct(owner.getKey(), Products.random());
+        PoolDTO pool1 = admin.owners().createPool(owner.getKey(), Pools.random(prod));
+        PoolDTO pool2 = admin.owners().createPool(owner.getKey(), Pools.random(prod));
+
+        ConsumerDTO consumer1 = admin.consumers().createConsumer(Consumers.random(owner));
+        ConsumerDTO consumer2 = admin.consumers().createConsumer(Consumers.random(owner));
+        admin.consumers().bindPool(consumer1.getUuid(), pool1.getId(), 1);
+        admin.consumers().bindPool(consumer2.getUuid(), pool2.getId(), 1);
+
+        ActivationKeyDTO key1 = admin.owners()
+            .createActivationKey(owner.getKey(), ActivationKeys.random(owner));
+        ActivationKeyDTO key2 = admin.owners()
+            .createActivationKey(owner.getKey(), ActivationKeys.random(owner));
+
+        key1 = admin.activationKeys().addPoolToKey(key1.getId(), pool1.getId(), 1L);
+        key2 = admin.activationKeys().addPoolToKey(key2.getId(), pool2.getId(), 1L);
+
+        assertThat(admin.entitlements().listAllForConsumer(consumer1.getUuid()))
+            .isNotNull()
+            .singleElement();
+
+        assertThat(admin.entitlements().listAllForConsumer(consumer2.getUuid()))
+            .isNotNull()
+            .singleElement();
+
+        assertThat(admin.activationKeys().getActivationKeyPools(key1.getId()))
+            .isNotNull()
+            .singleElement()
+            .returns(pool1.getId(), PoolDTO::getId);
+
+        assertThat(admin.activationKeys().getActivationKeyPools(key2.getId()))
+            .isNotNull()
+            .singleElement()
+            .returns(pool2.getId(), PoolDTO::getId);
+
+        // Update owner to SCA mode
+        owner.contentAccessMode(Owners.SCA_ACCESS_MODE);
+        owner = admin.owners().updateOwner(owner.getKey(), owner);
+
+        // Wait for the entitlement revoke job to finish
+        Thread.sleep(5000);
```

Comment:
instead of sleeping, you can use the listJobStatuses API with arguments being the owner key and the job key, which should return that single job status, then you can wait until the job is done, and assert on its end state. We have other spec tests (JobStatusSpecTest/CloudRegistrationSpecTest) that do this kind of thing

Reply:
On that note, there are some newer methods on the job assertion for this which we should be sure to use. Some examples of the newer logic is in the import spec tests, but as an example:

```
  AsyncJobStatusDTO importJob1 = this.importAsync(owner, manifest);
  assertThatJob(importJob1)
      .isFinished()
      .contains("SUCCESS");
```

Reply:
Updated

Reply:
We could also assert on the result of the job here. The result string should contain the amount of entitlements revoked and pools removed from activation keys

Reply:
Updated